src/vos/vos_tree.c

```diff
@@ -629,18 +628,17 @@ svt_rec_free_internal(struct btr_instance *tins, struct btr_record *rec,
 		return umem_free(&tins->ti_umm, rec->rec_off);
 
 	/** Find an empty slot for the deferred free */
-	for (i = dth->dth_rsrvd_cnt - 1; i >= 0; i--) {
-		dru = &dth->dth_rsrvds[i];
-		if (dru->dru_scm == NULL)
-			continue;
+	for (i = 0; i < dth->dth_deferred_cnt; i++) {
+		rsrvd_scm = dth->dth_deferred[i];
+		D_ASSERT(rsrvd_scm != NULL);
 
-		rsrvd_scm = dru->dru_scm;
 		if (rsrvd_scm->rs_actv_at >= rsrvd_scm->rs_actv_cnt)
 			continue; /* Can't really be > but keep it simple */
```

Comment:
The check is redundant, because each sub modification will call defer_free() at most once, "rs_actv_at" will be always zero.

Reply:
if it modifies multiple akeys, it can happen more than once, no?

Reply:
You are right. I ignored such case.
On the other hand, if some SCM area is not allocated by current current DTX, defer_free() it, whether will it cause issue or not? I will assume that it can work since I am not familiar with the semantics for defer_free().

Reply:
It will not cause an issue to defer free something not allocated by (or reserved for) the transaction.  However, this case only happens if the reservation has the same epoch and is in the same tx.